### PR TITLE
DE48038 - Fix Mobile dropdown tray opening transition in dialogs

### PR DESCRIPTION
**Pull request by svanherk**
So I still don't know exactly why. but this seems to fix both the problem of unsmooth opening, and being contained to the dialog height.  No dialog JS code runs on dropdown open, but it seems like the dialog transition reruns and applies itself to the dropdown's transition as well.  Switching it to instead use an animation with keyframes (like dropdown) has seemed to fix it.
Tested with both native and non-native in Chrome, Edge, Safari (Sauce) and Firefox (Sauce because IT has me locked on an older version with no native dialog) on the demo pages.

Is there an easy way to test this in an iframed FRA and/or the LMS?

Before:
mobile dropdown - broken

After:
mobile dropdown - fixed

### CODE REVIEW

File: components/dialog/dialog-styles.js [hunk 1]
```diff
@@ -45,10 +59,6 @@ export const dialogStyles = css`
 		z-index: 1000;
 	}
 
-	.d2l-dialog-outer.d2l-dialog-outer-nested {
-		top: 0;
```
[the remark this answers is not in the document]
Comment (svanherk) [reply components/dialog/dialog-styles.js:49]: So after making the other changes to `animation`, this was causing the nested child dialogs to open at the top of the window, not at the same height as the parent dialog.  I think this is related to how this change "magically fixed" the mobile dropdown height as well - before, `top: 0;` was the top of the parent dialog and now it's the somehow back to the top of the whole screen.  So we no longer need this override for nested dialogs, although I'd like to somehow verify this doesn't cause issues in iFRAs

[the remark this answers is not in the document]
Comment (annabelsegalld2l) [reply components/dialog/dialog-styles.js:49]: Gotcha - that makes sense.

**Comment by dbatiste** [comment]
To close to loop on part of the mystery here: the reason why changing from `transition` to `animation` fixes the height issue... from the [Mozilla docs on fixed position](https://developer.mozilla.org/en-US/docs/Web/CSS/position):
>The element is removed from the normal document flow, and no space is created for the element in the page layout. It is positioned relative to the initial [containing block](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block) established by the [viewport](https://developer.mozilla.org/en-US/docs/Glossary/Viewport), except when one of its ancestors has a transform, perspective, or filter property set to something other than none (see the [CSS Transforms Spec](https://www.w3.org/TR/css-transforms-1/#propdef-transform)), in which case that ancestor behaves as the containing block. (Note that there are browser inconsistencies with perspective and filter contributing to containing block formation.) Its final position is determined by the values of top, right, bottom, and left.

File: components/dialog/dialog-styles.js [hunk 2]
```diff
@@ -21,15 +21,29 @@ export const dialogStyles = css`
...
+	}
+
+	:host([_state="showing"]) > .d2l-dialog-outer {
+		/* must target direct child to avoid ancestor from interfering with closing child dialogs in Legacy-Edge */
+		animation: d2l-dialog-close 200ms ease-in;
+	}
+
+	@keyframes d2l-dialog-open {
+		0% { transform: translateY(0); }
+		100% { transform: translateY(-50px); }
```
[the remark this answers is not in the document]
Comment (svanherk) [reply components/dialog/dialog-styles.js:41]: Updated!

File: components/dialog/dialog-fullscreen.js [hunk 3]
```diff
@@ -161,10 +170,6 @@ class DialogFullscreen extends LocalizeCoreElement(AsyncContainerMixin(DialogMix
 					height: calc(100% - 1px);
 				}
 
-				div[nested].d2l-dialog-outer {
-					top: 0;
-				}
-
```
Comment (svanherk) [inline components/dialog/dialog-fullscreen.js:167]: As far as I can tell, this block and the corresponding one in `dialog.js` don't do anything.  `nested` used to be added as an attribute, but was changed to the `d2l-dialog-outer-nested` class in https://github.com/BrightspaceUI/core/pull/573.  I think these were just missed in the update, and switching to the `class` changed the specificity such that nothing broke and these `media` overrides weren't needed.

**Review by dbatiste — APPROVED** [review]

**Review by annabelsegalld2l — APPROVED** [review]
Wacky mystery resolved 🕵️ Thanks all

**Pull request merged**